src/main/kotlin/no/nav/tiltakspenger/saksbehandling/meldekort/service/overta/OvertaMeldekortBehandlingService.kt

```diff
@@ -0,0 +1,43 @@
+package no.nav.tiltakspenger.saksbehandling.meldekort.service.overta
+
+import arrow.core.Either
+import no.nav.tiltakspenger.libs.personklient.pdl.TilgangsstyringService
+import no.nav.tiltakspenger.saksbehandling.felles.exceptions.TilgangException
+import no.nav.tiltakspenger.saksbehandling.meldekort.domene.MeldekortBehandling
+import no.nav.tiltakspenger.saksbehandling.meldekort.domene.MeldekortBehandlingStatus
+import no.nav.tiltakspenger.saksbehandling.meldekort.ports.MeldekortBehandlingRepo
+
+class OvertaMeldekortBehandlingService(
+    private val tilgangsstyringService: TilgangsstyringService,
+    private val meldekortBehandlingRepo: MeldekortBehandlingRepo,
+) {
+    suspend fun overta(command: OvertaMeldekortBehandlingCommand): Either<KunneIkkeOvertaMeldekortBehandling, MeldekortBehandling> {
+        val meldekortBehandling = meldekortBehandlingRepo.hent(command.meldekortId)
+            ?: throw IllegalStateException("Fant ikke meldekortBehandling for id ${command.meldekortId}")
+        tilgangsstyringService.harTilgangTilPerson(
+            meldekortBehandling.fnr,
+            command.saksbehandler.roller,
+            command.correlationId,
+        )
+            .onLeft {
+                throw TilgangException("Feil ved tilgangssjekk til person ved overtakelse av meldekortbehandling. Feilen var $it")
+            }.onRight {
+                if (!it) throw TilgangException("Saksbehandler ${command.saksbehandler.navIdent} har ikke tilgang til person")
+            }
+
+        return meldekortBehandling.overta(command.saksbehandler).onRight {
+            when (it.status) {
```

Comment:
Vi skal ikke måtte sjekke statusen på nytt i servicen. Det bør gjøres i domeneklassen (og gjøres). Også bør den kun gi right når vi faktisk skal måtte lagre noe.

Reply:
Ser hva du mener sånn koden er nå, men når jeg legger til mulighet for å overta som beslutter må vi vel gjøre det sånn for å skille på om vi skal oppdatere for saksbehandler eller beslutter, tilsvarende som her? https://github.com/navikt/tiltakspenger-saksbehandling-api/blob/main/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/behandling/service/behandling/overta/OvertaBehandlingService.kt#L30

(alternativt kan jeg sjekke på typen meldekortbehandling, men noe må jeg sjekke på, eller så må vi ha en generisk update som tar inn og oppdaterer hele objektet i databasen, men det føles litt skummelt når vi har så store objekter med så mye logikk på seg 😅)